### PR TITLE
docs/feat: Add SDK page info, update components to be able to display Wallet SDK links

### PR DESCRIPTION
**Pull request by sprngbk**
## Description

This PR:
* Docs: Updates the SDKs page with [info on SDKs and how they can be used together](https://github.com/immutable/imx-docs/pull/86/files#diff-3370abe3b88b2e7baca73cf83060c2145191dc545d5c29adb16e69a93e25b65c)
* Feat: Adds a [category for wallet and link SDKs](https://github.com/immutable/imx-docs/pull/86/files#diff-3370abe3b88b2e7baca73cf83060c2145191dc545d5c29adb16e69a93e25b65cR93-R97) on SDKs page
* Feat: Add [wallet and link SDK constants](https://github.com/immutable/imx-docs/pull/86/files#diff-11bdf8babe536adbf336e8de7bdfe65b3a86d4d7011c51da852b69e8d3248af8R36-R60)
* Feat: Update switcher to [take arguments](https://github.com/immutable/imx-docs/pull/86/files#diff-8a45a89ce21c95fa0c2fbb4a1a46c19364f9e3dcbfd4836a53b86c1168c00c84R20) so that both wallet and core SDK details can be passed to it

TODO:
- [x] Pass Wallet SDK constants to Switcher so that Wallet SDKs can be linked to from there (may have to update some logic)
- [x] Wallets team to add Wallet SDK pages so that links on SDK page don't go to empty pages

## Resolved issues
Jira: N/A

### Before submitting the PR, please consider the following:
- [x] It's beneficial if your pull request references an issue used to discuss the pull request ahead of time. If you haven't previously created an issue, please create one and discuss your contribution with the maintainers.
- [x] Prefix your PR title with `feat: `, `fix: `, `chore: `, `docs:`, or `refactor:`.
- [x] The description should clearly illustrate what problems the pull request solves.
- [x] Ensure that the commit messages follow our guidelines.
- [x] Resolve merge conflicts (if any).
- [x] Make sure that the current branch is up to date with the `main` branch.

### For internal Immutable developers:
- [x] If you are adding or updating a documentation page, please also add or update the team ownership of that page (follow [this guide](https://immutable.atlassian.net/wiki/spaces/PPS/pages/1947927045/Docs+ownership+registry#What-to-do-when-adding%2Fupdating-documentation-pages%3F)).
- [ ] If you are adding or updating documentation for an SDK, please make sure that you meet the requirements in [this doc](https://immutable.atlassian.net/wiki/spaces/PPS/pages/1916994017/SDK+documentation+guide).
- [ ] Please obtain PR approval from all of the following:
    - Your tech lead (Fallback: Team lead or engineering manager)
    - Your product manager (Fallback: Group PM)
    - Another software engineer on your team

### CODE REVIEW

**Review by RenatoChitolina — APPROVED** [review]

File: docs/main/sdks.mdx [hunk 1]
```diff
@@ -3,15 +3,99 @@ id: 'sdks'
...
+
+
+## Core SDKs
+<SdkButtonList sdks={coreSDKs} />
+
+## Wallet SDKs
+<SdkButtonList sdks={walletSDKs} />
+
+## Link SDK
+<SdkButtonList sdks={linkSDK} sdks={linkSDK} />
```
Comment (matthewmuscat) [inline docs/main/sdks.mdx:98]: Duplicate props?
```suggestion
<SdkButtonList sdks={linkSDK} />
```

File: src/components/SdkSwitcher/index.tsx [hunk 2]
```diff
@@ -6,21 +6,14 @@ import { Listbox, Transition } from '@headlessui/react';
...
-  return localId;
-};
-
-const SdkSwitcher = () => {
+const SdkSwitcher = ({ sdks }) => {
   const history: History = useHistory();
-  const sdkId = getSdkId(sdks);
-  const [selectedSdk, setSelectedSdk] = useState<SdkItem>(sdks[sdkId]);
+  const currentSDK = sdks.find(
+    (sdk: { url: string }) => sdk.url == history.location.pathname
```
Comment (matthewmuscat) [inline src/components/SdkSwitcher/index.tsx:14]: Can you just `url == history.location.pathname` here as you've destructured `url`?

**Review by matthewmuscat — CHANGES_REQUESTED** [review]

File: src/components/SdkSwitcher/index.tsx (hunk 2 above)
Comment (sprngbk) [reply src/components/SdkSwitcher/index.tsx:14]: Oh oops, it's not destructured. It's just specifying the type for `sdk`. I think I'll leave it un-destructured for clarity?

**Review by matthewmuscat — APPROVED** [review]

File: src/components/SdkSwitcher/index.tsx [hunk 3]
```diff
@@ -6,21 +6,14 @@ import { Listbox, Transition } from '@headlessui/react';
...
+  );
+  const [selectedSdk, setSelectedSdk] = useState<SdkItem>(currentSDK);
 
   const handleOnChange = (sdkItem: SdkItem) => {
     localStorage.setItem(SdkIdKey, sdkItem.id.toString());
```
Comment (CodeSchwert) [inline src/components/SdkSwitcher/index.tsx:19]: ✅

**Pull request merged**